:sparkles: - pdf merge now works on arrays of length 1

JLuzz opened the pull request:
I think you should reconsider throwing an error in the case the array is equal to length 1

Although it makes no difference just merging one document, if the case ever does come up it is better to just let it run instead of throwing an error

For example, the code I am working with the input to pdf merge is based on the size of the users document. If their document only has 1 page pdf-merge will throw an error

JLuzz commented:
It is the latter. You are correct in stating that a merge with one entity with nothingness resolves the single entity, so it is wasteful to use system resources. However, I don't think that the case of 1 pdf should be treated as an error.

pdf-merge would be more flexible if it accepts the case of 1 and returns the original document or the expected output format, like you said. I would be happy to write up a PR that handles this case without firing up pdftk

wubzz commented:
You're welcome to do so. Any solution that does not involve calling pdftk is ok with me.

wubzz commented on index.js:
```diff
@@ -30,10 +30,13 @@ module.exports = (files, options) => new Promise((resolve, reject) => {
+      return output(buffer);
+    })
+    .then(resolve)
+    .catch(reject);
```
You need a return here as well. Right now it will still call pdftk as well.

JLuzz commented:
Along with adding the return statement I also changed the error cases to one-liners, let me know what you think. I can change them back if you don't like them

JLuzz commented:
Once I added the return statements, output became undefined for the files.length === 1 case, so I moved both options and output to be declared before that case occurs. Is there a better way to organize it?

wubzz commented:
@JLuzz I much rather prefer keeping and using brackets like `if(length > 1) {...}` rather than `if(length > 1) ...`.

What I meant in my previous comment was the following:

```javascript
if(files.length === 1) {
    readFile(files[0])
    .then((buffer) => {
      return resolve(output(buffer));
    })
    .catch(reject);

      return;
}
```

Adding a return **after** the promise chain prevents the rest of the function being executed, which in turn means you won't need the new if statement `if(files.length > 1)` below which is also always going to be true if we reach that point of the code.

JLuzz commented:
Thank you for the clarification

wubzz commented:
Thanks, looks good now
